src/NBitcoin/BitcoinAddress.cs

```diff
@@ -18,23 +18,30 @@ public BitcoinScriptAddress(string base58, Network expectedNetwork)
 
         private static string Validate(string base58, ref Network expectedNetwork)
         {
-            if(base58 == null)
+            if (IsValid(base58, ref expectedNetwork))
+                return base58;
+            throw new FormatException("Invalid BitcoinScriptAddress");
+        }
+
+        public static bool IsValid(string base58, ref Network expectedNetwork)
```

Comment:
This is a bit confusing, as we pass in a network and use it in determining validity then change it if we find a different valid network.

Reply:
Agree. Refactored to only validate with the expectedNetwork param & no loop